File: src/runtime/dart/std/os_native.dart
// std/os_native.dart — stub native for pytra.std.os
// source: src/runtime/dart/std/os_native.dart

import 'dart:io';

String getcwd() => Directory.current.path;

void mkdir(dynamic path, [dynamic existOk]) {
  bool eo = (existOk == true);
  try {
    Directory(path.toString()).createSync(recursive: false);
  } catch (e) {
    if (!eo) rethrow;
  }
}

void makedirs(dynamic path, [dynamic existOk]) {
  bool eo = (existOk == true);
  try {
    Directory(path.toString()).createSync(recursive: true);
  } catch (e) {
    if (!eo) rethrow;
  }
}

File: src/runtime/zig/std/os_native.zig
// Native implementation of pytra.std.os for Zig.
// source: src/pytra/std/os.py

const std = @import("std");
pub const path = @import("os_path_native.zig");

pub fn getcwd() []const u8 {
    // Stub: return "." for now
    return ".";
}

pub fn mkdir(p: []const u8) void {
    std.fs.cwd().makeDir(p) catch {};
}

pub fn makedirs(p: []const u8, exist_ok: bool) void {
    _ = exist_ok;
    std.fs.cwd().makePath(p) catch {};
}
